src/Microsoft.ML.Data/DataLoadSave/DataOperationsCatalog.cs

```diff
@@ -433,6 +433,60 @@ public TrainTestData TrainTestSplit(IDataView data, double testFraction = 0.1, s
             return new TrainTestData(trainFilter, testFilter);
         }
 
+        /// <summary>
+        /// Split the dataset into cross-validation folds of train set and test set.
+        /// Respects the <paramref name="samplingKeyColumnName"/> if provided.
+        /// </summary>
+        /// <param name="data">The dataset to split.</param>
+        /// <param name="numberOfFolds">Number of cross-validation folds.</param>
+        /// <param name="samplingKeyColumnName">Name of a column to use for grouping rows. If two examples share the same value of the <paramref name="samplingKeyColumnName"/>,
+        /// they are guaranteed to appear in the same subset (train or test). This can be used to ensure no label leakage from the train to the test set.
+        /// If <see langword="null"/> no row grouping will be performed.</param>
+        /// <param name="seed">Seed for the random number generator used to select rows for cross-validation folds.</param>
+        /// <example>
+        /// <format type="text/markdown">
+        /// <![CDATA[
+        /// [!code-csharp[TrainTestSplit](~/../docs/samples/docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/CrossValidationSplit.cs)]
+        /// ]]>
+        /// </format>
+        /// </example>
+        public List<TrainTestData> CrossValidationSplit(IDataView data, int numberOfFolds = 5, string samplingKeyColumnName = null, int? seed = null)
```

Comment:
>List [](start = 15, length = 4)

`IEnumerable` or `IList` so we have less to hold onto?

Reply:
I have better idea!

---
In reply to: [267986740](https://github.com/dotnet/machinelearning/pull/3058#discussion_r267986740) [](ancestors = 267986740)